sdks/python/apache_beam/io/vcfio.py

```diff
@@ -0,0 +1,378 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""A source for reading from VCF files (version 4.x).
+
+The 4.2 spec is available at https://samtools.github.io/hts-specs/VCFv4.2.pdf.
+"""
+
+from __future__ import absolute_import
+import collections
+
+from apache_beam.coders import coders
+from apache_beam.io.filesystem import CompressionTypes
+from apache_beam.io.iobase import Read
+from apache_beam.io import filebasedsource
+from apache_beam.io.textio import _TextSource as TextSource
```

Comment:
Oh! I didn't realize that TextSource wasn't public at all. Please do not import the private implementation details of other modules - they, again, are subject to change without any notice. The proper way to address this here would be to make necessary generalizations to the (public) ReadFromText transform, and express ReadFromVcf as a composite transform wrapping ReadFromText - rather than use TextSource directly.

Reply:
I looked at the design doc and seems that @chamikaramj was fine with this. Okay then.